pkg/dataplane/controller.go

```diff
@@ -310,6 +329,98 @@ func (c *Controller) evaluatePacket(ctx context.Context, p *network.Packet) bool
 	return allowed
 }
 
+// syncNFTablesRules adds the necessary rules to process the first connection packets in userspace
+// and check if network policies must apply.
+// TODO: We can divert only the traffic affected by network policies using a set in nftables or an IPset.
+func (c *Controller) cleanConntrackEntries(ctx context.Context) error {
+	logger := klog.FromContext(ctx)
+
+	logger.Info("Cleaning conntrack entries")
+	start := time.Now()
+	defer func() {
+		logger.Info("Cleaning conntrack entries", "elapsed", time.Since(start))
+	}()
+
+	flows, err := vishnetlink.ConntrackTableList(vishnetlink.ConntrackTable, vishnetlink.FAMILY_ALL)
+	if err != nil {
+		return err
+	}
+
+	allPodIPs, divertAll, err := c.policyEngine.GetManagedIPs(ctx)
+	if err != nil {
+		return err
+	}
+
+	ipset := sets.Set[string]{}
+	for _, ip := range allPodIPs {
+		ipset.Insert(ip.String())
+	}
+
+	filter := newConntrackFilter()
+
+	for _, flow := range flows {
+		// TODO: only evaluate the flows that are established
+		// TCP for now, as they are the only ones supported by the library.
+		// UDP is not supported yet, but it can be done by parsing the raw attributes
+		// new flows will be evaluated by the nfqueue logic.
+		if flow.ProtoInfo == nil {
+			continue
+		}
+		state, ok := flow.ProtoInfo.(*vishnetlink.ProtoInfoTCP)
+		if !ok {
+			continue
+		}
+		if state.State != nl.TCP_CONNTRACK_ESTABLISHED {
+			continue
+		}
+		// If divertAll is true, all pod IPs are managed by network policies.
+		if !divertAll {
+			// Only evaluate flows that are affected by network policies.
+			// Translated packets will be evaluated on the destination node.
+			if !ipset.Has(flow.Forward.SrcIP.String()) || !ipset.Has(flow.Forward.DstIP.String()) {
+				continue
+			}
+		}
+
+		packet := generatePacket(flow)
+		if packet == nil {
+			continue
+		}
+
+		allowed, err := c.policyEngine.EvaluatePacket(ctx, packet)
+		if err != nil {
+			klog.Infof("error evaluating conntrack entry %v: %v", flow, err)
+			continue
+		}
+		// If the flow is not allowed, add it to the filter for deletion.
+		if !allowed {
+			filter.addFlow(flow)
+		}
+
+	}
+	var errorList []error
+	if len(filter.flowsV4) > 0 {
+		logger.V(2).Info("Cleaning IPv4 conntrack entries", "entries", len(filter.flowsV4))
+		n, err := vishnetlink.ConntrackDeleteFilters(vishnetlink.ConntrackTable, unix.AF_INET, filter)
+		if err != nil {
+			errorList = append(errorList, err)
+		}
+		logger.V(4).Info("Deleted conntrack entries", "entries", n)
+	}
```

Comment:
It's weird how `filter.addFlow()` hides IPv4-vs-IPv6 but then you have to think about it here... It seems like it would make more sense to move more of this into `conntrack.go`

Reply:
conntrafk flush does not work, conntrack has some memory that recovers the state :/